### DISCUSSION TITLE
Bots from CTFNote events

### DISCUSSION THREAD

**Discussion by XeR** [TFNS/CTFNote#97] [Ideas]
As discussed in #53, we need to consider a way to add bots.

What's the best way to do this ?
I think we could add web hooks to the config file and have CTFNote send events to HTTP endpoints.

I think most teams use [Discord](https://discord.com/developers/docs/resources/webhook), but I know some teams use Slack and Mattermost.

How do we handle events' sensitivity ?
Do we send every flags to the webhook ? Only new tasks ? Do we want this to be configurable ?

**Comment by JJ-8** [comment]
Discord bot is now included in v3.0.0 but I also opened an issue for using different styles of handling the channels: https://github.com/TFNS/CTFNote/issues/239.
I think it is best to generalize the current Discord implementation to a plugin system and then we can add different Discord bot plugins which handles task channels differently. This would also open the possibility to implement Slack and Mattermost bots.
